app/src/main/java/org/kiwix/kiwixmobile/webserver/ZimHostActivity.java

```diff
@@ -0,0 +1,473 @@
+package org.kiwix.kiwixmobile.webserver;
+
+import android.Manifest;
+import android.app.Activity;
+import android.app.ProgressDialog;
+import android.content.ComponentName;
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentSender;
+import android.content.ServiceConnection;
+import android.content.pm.PackageManager;
+import android.net.ConnectivityManager;
+import android.net.wifi.WifiConfiguration;
+import android.os.Build;
+import android.os.Handler;
+import android.os.IBinder;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+import android.widget.TextView;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.annotation.RequiresApi;
+import androidx.appcompat.app.AlertDialog;
+import androidx.appcompat.app.AppCompatActivity;
+import android.os.Bundle;
+import androidx.appcompat.widget.Toolbar;
+import androidx.core.app.ActivityCompat;
+import androidx.core.content.ContextCompat;
+import androidx.fragment.app.FragmentManager;
+import androidx.fragment.app.FragmentTransaction;
+import butterknife.BindView;
+import com.google.android.gms.common.api.ApiException;
+import com.google.android.gms.common.api.ResolvableApiException;
+import com.google.android.gms.location.LocationRequest;
+import com.google.android.gms.location.LocationServices;
+import com.google.android.gms.location.LocationSettingsRequest;
+import com.google.android.gms.location.LocationSettingsResponse;
+import com.google.android.gms.location.LocationSettingsStates;
+import com.google.android.gms.location.LocationSettingsStatusCodes;
+import com.google.android.gms.tasks.Task;
+import java.lang.reflect.Method;
+import org.kiwix.kiwixmobile.R;
+import org.kiwix.kiwixmobile.wifi_hotspot.HotspotService;
+import org.kiwix.kiwixmobile.zim_manager.fileselect_view.ZimFileSelectFragment;
+
+import static org.kiwix.kiwixmobile.utils.StyleUtils.dialogStyle;
+import static org.kiwix.kiwixmobile.webserver.WebServerHelper.getAddress;
+import static org.kiwix.kiwixmobile.webserver.WebServerHelper.isServerStarted;
+
+public class ZimHostActivity extends AppCompatActivity implements
```

Comment:
This activity should subclass `BaseActivity` not `AppCompatActivity`.

Reply:
Fixed. e8b2cf4eacc708ea37af08c73bdfb41f78b861d8

Reply:
Not fixed. This activity has to be added to **ActivityBindingModule**. We are getting a crash when we open it because of this. Try running the code before committing it.

Reply:
`  @PerActivity
  @ContributesAndroidInjector
  public abstract ZimHostActivity provideZimHostActivity();
`
Would this work?

Reply:
Yeah. You should try locally.

Reply:
Works fine. c0c67e5de91fe1